lib/protocol/http1/connection.rb

```diff
@@ -171,6 +171,8 @@ def write_headers(headers)
 			
 			def read_line?
 				@stream.gets(CRLF, chomp: true)
+			rescue IOError, Errno::ECONNRESET
```

Comment:
Can you rebase on master. Not sure if we need this on the latest version or not. Thoughts?